ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -1,33 +1,28 @@
-//! Yk's AOT IR deserialiser.
+//! The AOT Intermediate Representation (IR).
 //!
-//! This module contains the data structures for the AOT IR and a parser to read it from its
-//! serialised format.
+//! This is the IR created by ykllvm: this module contains both the IR itself and a deserialiser
+//! from ykllvm's output into the IR itself. The IR can feel a little odd at first because we store
+//! indexes into vectors rather than use direct references. This allows us to squeeze the amount of
+//! the memory used down (and also bypasses issues with representing graph structures in Rust, but
+//! that's slightly accidental).
 //!
-//! The AOT IR accurately reflects the structure and semantics of the AOT binary. As such, it must
-//! not be mutated after the fact.
+//! The IR uses two general terminological conventions:
+//!   * A "definition" is something for which the IR contains complete knowledge.
+//!   * A "declaration" is something compiled externally for which we only know the minimal
+//!   external structure (e.g. a function signature), with the body being present elsewhere in the
```

Comment:
Another optional nitpick: For variable declarations, you'd have an "initialiser" not a "body". Bodies only really make sense when you talk about functions.

Reply:
I did wonder about this, but couldn't think of a term that is a superset of these. I think it's kind-of clear as-is, but am happy to take a suggestion for an improvement.

Reply:
Probably not worth worrying about at this early stage.